### PR TITLE
SAK-33706 order by submission id on anonymous assignments

### PR DESCRIPTION
**Pull request by bgarciaentornos**

### CODE REVIEW

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 1]
```diff
@@ -1639,13 +1640,18 @@ public void getSubmissionsZip(OutputStream out, String reference, String query)
                     List<AssignmentSubmission> submissions = new ArrayList<AssignmentSubmission>(submitters.values());
 
                     StringBuilder exceptionMessage = new StringBuilder();
-
+                    SortedIterator sortedIterator;
+                    if (assignmentUsesAnonymousGrading(assignment)){
+                        sortedIterator = new SortedIterator(submissions.iterator(), new AnonymousSubmissionComparator());
```
Comment (ern) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: This should cause some type of exception as there is a private no arg constructor, meaning should never instantiate this class using the noarg constructor.

Comment (bgarciaentornos) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: But this class does not have a private no arg constructor, it's the AssignmentSubmissionComparator one.

Comment (ern) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java]: thanks for the clarification!

**Comment by ern** [comment]
need to resolve conflicts

**Pull request merged**